### PR TITLE
Rewrite existing docs instead of adding a new section

### PR DESCRIPTION
**Pull request by schmichael**
@xiang90 pointed out my earlier PR duplicated a lot of things that
were mentioned earlier in the doc.

This time around I tried just making some gotchas more explicit in the
existing docs instead of just tacking new stuff onto the end.

(Previous attempt: #2776)

### CODE REVIEW

File: Documentation/api.md [hunk 1]
```diff
@@ -277,27 +277,29 @@ The first terminal should get the notification and return with the same response
 However, the watch command can do more than this.
 Using the index, we can watch for commands that have happened in the past.
 This is useful for ensuring you don't miss events between watch commands. 
-Typically, we watch again from the (modifiedIndex + 1) of the node we got.
+Typically, we watch again from the `modifiedIndex + 1` of the node we got.
 
-Let's try to watch for the set command of index 7 again:
```
Comment (xiang90) [inline Documentation/api.md]: um... actually i still want to say user can get the previous event immediately. @yichengq @barakmich  any opinion?

Comment (yichengq) [inline Documentation/api.md]: I think both examples(wait on index 7 & index 8) are valuable.

**Comment by xiang90** [comment]
ping @schmichael. Can you please revisit this when you have time? Thanks!

**Comment by xiang90** [comment]
ping @schmichael

File: Documentation/api.md (hunk 1 above)
Comment (schmichael) [inline Documentation/api.md]: Fixing, sorry for the massive delay.

**Comment by schmichael** [comment]
Sorry for the massive delay. Updated.

**Comment by xiang90** [comment]
@yichengq Can you please take a look?

File: Documentation/api.md [hunk 2]
```diff
@@ -325,13 +335,16 @@ curl 'http://127.0.0.1:2379/v2/keys/foo' -vv
...
-```sh
-curl 'http://127.0.0.1:2379/v2/keys/foo?wait=true&waitIndex=2003'
-```
+1. The `X-Etcd-Index` will be greater than or equal to the `modifiedIndex`
+2. If it is greater than the `modifiedIndex`, none of those indexes will be
+   related to the key being fetched.
 
+Using the `modifiedIndex` will be functionally equivalent for subsequent
+watches, but since it might be smaller than the `Etcd-Index` we may receive a
+`401` error more quickly than if we used the `Etcd-Index`.
```
Comment (yichengq) [inline Documentation/api.md]: could we give a expected curl command here? So it can be more straight-forward.

**Comment by yichengq** [comment]
lgtm. I wanna merge this one. This PR does improve readability much, and i will send following PR to address my own concerns.
/cc @xiang90

**Comment by xiang90** [comment]
LGTM

**Comment by yichengq** [comment]
I stopped to merge this one because I found its commit title doesn't have package name. I cherry-pick these commits into a new PR #3128. Let us discuss in that one.

**Pull request closed without merging**